### PR TITLE
Feature/test-cloud espresso support

### PR DESCRIPTION
**Pull request by krukow**
A first take for Espresso support for Test beacon

@kmiecikt please review this draft proposal -- I'm aware of quite a bit of duplication, but I wanted to have you own how you want code to be shared

Note: also created a `./mobile-center` helper for OS X which I found useful as a beginner

Note: requires https://github.com/xamarin/test-cloud-frontend/commit/44015bd66a8838c24dd11a2c8a1c9a5d2163dd58 to test this end to end (this should be deployed tomorrow at least to staging)

### CODE REVIEW

File: src/commands/test/lib/espresso-preparer.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+      this.buildDir,
+      false,
+      `Espresso build directory "${this.buildDir}" doesn't exist`);
+  }
+
+  private async validateTestApkExists(): Promise<void> {
+    await this.testApkPath();      
+  }
+
+  private async testApkPath(): Promise<string> {
```
Comment (kmiecikt) [inline src/commands/test/lib/espresso-preparer.ts]: The "test prepare uitest" requires "--app-path" option that points to the application file. Should we do the same here?

Comment (krukow) [inline src/commands/test/lib/espresso-preparer.ts]: yeah `--app-path` is definitely required, and yes, I think we should allow (like uitest tool dir option) the user to override the espresso tests apk file with an option. Thanks for that.

**Comment by krukow** [comment]
@kmiecikt addressed your review comments - please re-review.

Successfully uploaded espresso test:

https://mobile.azure.com/users/kakrukow-qggr/apps/squadgoals/test/runs/e78e8a0d-6da3-4617-8949-1cc8472bb0e0

https://testcloud.xamarin.com/test/e78e8a0d-6da3-4617-8949-1cc8472bb0e0/pipeline 


Commands: 
with  `--test-apk-path`
```
./mobile-center test run espresso --app "kakrukow-qggr/squadgoals" \ 
     --devices 108b848e  \ 
    --app-path ~/code/squad-goals-android/app/build/outputs/apk/app-debug.apk \
    --test-series "master" --locale "en_US" \ 
    --test-apk-path ~/code/squad-goals-android/app/build/outputs/apk/app-debug-androidTest.apk
```
with  `--build-dir`
```
./mobile-center test run espresso --app "kakrukow-qggr/squadgoals" \ 
     --devices 108b848e  \ 
    --app-path ~/code/squad-goals-android/app/build/outputs/apk/app-debug.apk \
    --test-series "master" --locale "en_US" \ 
    --build-dir ~/code/squad-goals-android/app/build/outputs/apk
```

File: mobile-center [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
```
Comment (christav) [inline mobile-center]: It's been removed and replaced by the auto-generated npm launch script. I'd rather remove the ad-hoc launch scripts from the repo. For dev purposes we can take the hit to run node/ts-node directly, but it's less confusing for our customers, and easier for us to maintain installation/package publishing, if we have one canonical way to launch the thing.

**Comment by christav** [comment]
@krukow What's the status of this PR? Is it intended to be merged now, or is it still a sample/prototype?

**Comment by krukow** [comment]
@christav thanks. I was waiting for a final signoff from @kmiecikt but it's ready to go from my perspective.

File: mobile-center (hunk 2 above)
Comment (krukow) [inline mobile-center]: removed

**Pull request merged**